h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java

```diff
@@ -222,6 +223,25 @@ public XGBoostModel(Key<XGBoostModel> selfKey, XGBoostParameters parms, XGBoostO
     model_info = new XGBoostModelInfo(parms, dinfo);
   }
 
+  @Override
+  public void computeEffectiveParameters() {
+    EffectiveParametersUtils.initStoppingMetric(_parms, _effective_parms, _output.isClassifier(), _output.isAutoencoder());
+    EffectiveParametersUtils.initCategoricalEncoding(_parms, _effective_parms, _output.nclasses(), Parameters.CategoricalEncodingScheme.OneHotInternal);
+    EffectiveParametersUtils.initFoldAssignment(_parms, _effective_parms);
+    EffectiveParametersUtils.initDistribution(_parms, _effective_parms, _output.nclasses());
+    _effective_parms._backend = getActualBackend(_parms);
+    _effective_parms._dmatrix_type = _output._sparse ? XGBoostModel.XGBoostParameters.DMatrixType.sparse : XGBoostModel.XGBoostParameters.DMatrixType.dense;  
```

Comment:
this looks like a copy paste from XGBoost.java could we move the methods here rather and call them from here?